analytics-datastore-clickhouse/docker-compose.cluster.yml

```diff
@@ -2,160 +2,108 @@ version: '3.9'
-      - target: /etc/clickhouse-server/config.d/remote_servers.xml
-        source: clickhouse_remote_servers.xml
-      - target: /etc/clickhouse-server/config.d/use_keeper.xml
-        source: clickhouse_use_keeper.xml
+    depends_on:
+      - clickhouse-keeper-01
+      - clickhouse-keeper-02
+      - clickhouse-keeper-03
     networks:
       public:
       default:
 
   analytics-datastore-clickhouse-03:
-    image: clickhouse/clickhouse-server
+    image: clickhouse/clickhouse-server:latest
+    user: "101:101"
+    container_name: analytics-datastore-clickhouse-03
     hostname: analytics-datastore-clickhouse-03
     deploy: 
       placement: 
         constraints: 
           - "node.labels.name==node-3"
-    ulimits:
-      noFile: 262144
     volumes:
-      - clickhouse-data-03:/var/lib/clickhouse/        
-    configs:
-      - target: /etc/clickhouse-server/config.d/docker_related_config.xml
-        source: docker_related_config.xml
-      - target: /etc/clickhouse-server/config.d/enable_keeper.xml
-        source: clickhouse_enable_keeper_03.xml
-      - target: /etc/clickhouse-server/config.d/macros.xml
-        source: clickhouse_macros_03.xml
-      - target: /etc/clickhouse-server/config.d/remote_servers.xml
-        source: clickhouse_remote_servers.xml
-      - target: /etc/clickhouse-server/config.d/use_keeper.xml
-        source: clickhouse_use_keeper.xml
+      - ${PWD}/configs/clickhouse-03/etc/clickhouse-server/config.d/config.xml:/etc/clickhouse-server/config.d/config.xml
+      - ${PWD}/configs/clickhouse-03/etc/clickhouse-server/users.d/users.xml:/etc/clickhouse-server/users.d/users.xml
+      - clickhouse-data-03:/var/lib/clickhouse/
+    depends_on:
+      - clickhouse-keeper-01
+      - clickhouse-keeper-02
+      - clickhouse-keeper-03
     networks:
       public:
       default:
 
-  analytics-datastore-clickhouse-04:
-    image: clickhouse/clickhouse-server
-    hostname: analytics-datastore-clickhouse-04
-    ulimits:
-      noFile: 262144
+  clickhouse-keeper-01:
+    image: clickhouse/clickhouse-keeper:latest-alpine
+    user: "101:101"
+    container_name: clickhouse-keeper-01
+    hostname: clickhouse-keeper-01
+    deploy: 
+      placement: 
+        constraints: 
+          - "node.labels.name==node-1"
     volumes:
-      - clickhouse-data-04:/var/lib/clickhouse/        
-    configs:
-      - target: /etc/clickhouse-server/config.d/docker_related_config.xml
-        source: docker_related_config.xml
-      - target: /etc/clickhouse-server/config.d/macros.xml
-        source: clickhouse_macros_04.xml
-      - target: /etc/clickhouse-server/config.d/remote_servers.xml
-        source: clickhouse_remote_servers.xml
-      - target: /etc/clickhouse-server/config.d/use_keeper.xml
-        source: clickhouse_use_keeper.xml
-    networks:
-      public:
-      default:
+     - ${PWD}/configs/clickhouse-keeper-01/etc/clickhouse-keeper/keeper_config.xml:/etc/clickhouse-keeper/keeper_config.xml
```

Comment:
We can't use regular file mount with docker swarm as these files need to be available across all nodes. That is why we use docker config which will make the config available across all nodes.